README.md

```diff
@@ -99,41 +136,71 @@ into the element. This is the only method that must be implemented by subclasses
- for detailed API info, here are some highlights. Note, the leading underscore
- is used to indicate that these methods are
- [protected](https://en.wikipedia.org/wiki/Class_(computer_programming)#Member_accessibility);
- they are not private and can and should be implemented by subclasses.
- These methods generally are called as part of the rendering lifecycle and should
- not be called in user code unless otherwise indicated.
-
-  * `_createRoot()`: Implement to customize where the
+ for detailed API info, here are some highlights.
+
+  * `createRenderRoot()` (protected): Implement to customize where the
   element's template is rendered by returning an element into which to
   render. By default this creates a shadowRoot for the element.
   To render into the element's childNodes, return `this`.
 
-  * `_firstRendered()`: Called after the element DOM is rendered for the first time.
-
-  * `_shouldRender(props, changedProps, prevProps)`: Implement to control if rendering
-  should occur when property values change or `invalidate` is called.
+  * `shouldUpdate(changedProperties)` (protected): Implement to control if updating and rendering
+  should occur when property values change or `invalidate` is called. The `changedProps`
+  argument is an object with keys for the changed properties pointing to their previous values.
   By default, this method always returns true, but this can be customized as
-  an optimization to avoid rendering work when changes occur which should not be rendered.
-
-  * `_render(props)`: Implement to describe the element's DOM using `lit-html`. Ideally,
-  the `_render` implementation is a pure function using only `props` to describe
-  the element template. This is the only method that must be implemented by subclasses.
-
-  * `_didRender(props, changedProps, prevProps)`: Called after element DOM has been rendered.
-  Implement to directly control rendered DOM. Typically this is not needed as `lit-html`
-  can be used in the `_render` method to set properties, attributes, and
-  event listeners. However, it is sometimes useful for calling methods on
-  rendered elements, for example focusing an input:
-  `this.shadowRoot.querySelector('input').focus()`.
-
-  * `renderComplete`: Returns a promise which resolves after the element next renders.
-
-  * `_requestRender`: Call to request the element to asynchronously re-render regardless
+  an optimization to avoid updating work when changes occur, which should not be rendered.
+
+  * `update()` (protected): This method calls `render()` and then uses `lit-html` to
+  render the template DOM. Override to customize how the element renders DOM. Note,
+  during `update()` setting properties does not trigger `invalidate()`, allowing
+  property values to be computed and validated.
+
+  * `render()` (protected): Implement to describe the element's DOM using `lit-html`. Ideally,
+  the `render` implementation is a pure function using only the element's current properties
```

Comment:
link to description of what a pure function is, or describe it shortly

Reply:
Done